R/volcanoPlot.R

```diff
@@ -114,244 +108,381 @@ volcanoPlot <- function(contrastDF,
+
 
         # Optional Decorations
         if (sizeByIntensity) {
             volcanoPlot <- volcanoPlot + aes(size = LogInt) +
                 scale_size_continuous()
+        } else {
+            volcanoPlot <- volcanoPlot + aes(size = Group) +
+                scale_size_manual(name = "Group", guide = "legend", labels = ssc$group,
+                                  values = ssc$symbolSize)
         }
 
-        if (!is.null(pthresholdLine)) {
+        if (!is.null(referenceLine)) {
             volcanoPlot <- volcanoPlot +
-                geom_hline(yintercept = -log10(pthreshold), color = pthresholdLine,
-                           alpha = 0.5, size = refLineThickness)
+                geom_vline(xintercept = 0,
+                           color = referenceLine,
+                           size = refLineThickness,
+                           alpha = transparency)
         }
 
         if (!is.null(foldChangeLines)) {
             volcanoPlot <- volcanoPlot +
-                geom_vline(xintercept = foldChangeLines, color = symbolFill["Increased"],
-                           alpha = 0.5, size = refLineThickness) +
-                geom_vline(xintercept = -foldChangeLines, color = symbolFill["Decreased"],
-                           alpha = 0.5, size = refLineThickness)
+                geom_vline(xintercept = foldChangeLines,
+                           color = colors["Increased"],
+                           size = refLineThickness,
+                           alpha = transparency) +
+                geom_vline(xintercept = -foldChangeLines,
+                           color = colors["Decreased"],
+                           size = refLineThickness,
+                           alpha = transparency)
         }
 
-        # Add geneSym labels to increased & decreased genes
+        # Add genesym labels to increased, decreased genes
         if (!missing(geneSymLabels) && !missing(geneSymCol)) {
-            # Filter contrastDF to changed genes
-            idx <- contrastDF[[geneSymCol]] %in% geneSymLabels
-            contrastDFsubset <- contrastDF[idx,]
+            labeldata <- contrastDF[[geneSymCol]] %in% geneSymLabels
+            contrastDF_labelsubset <- contrastDF[labeldata,]
             volcanoPlot <- volcanoPlot +
-                geom_text_repel(data = contrastDFsubset, aes_string(x = x, y = y, label = geneSymCol),
-                                show.legend = FALSE)
+                ggrepel::geom_text_repel(data = contrastDF_labelsubset,
+                                aes_string(x = logRatioCol, y = "negLog10P", label = geneSymCol),
+                                show.legend = TRUE, max.overlaps = dim(contrastDF_labelsubset)[1]*10)
         }
 
-        # Add Labels
-        volcanoPlot <- volcanoPlot +
-            xlab(xlab) +
-            ylab(ylab) +
-            ggtitle(title)
-
-        volcanoPlot <- setLegendPosition(volcanoPlot, legendPosition)
+        # Add axis Labels
```

Comment:
Value assignment has to be done right at the top along with validating xlab and ylab - before even go into defining plot functions.